flycheck.el

```diff
@@ -3395,9 +3408,37 @@ Return the created overlay."
       (setf (overlay-get overlay 'before-string)
             (flycheck-error-level-make-fringe-icon
              level flycheck-indication-mode)))
-    (setf (overlay-get overlay 'help-echo) (flycheck-error-message err))
+    (setf (overlay-get overlay 'help-echo) #'flycheck-help-echo-helper)
     overlay))
 
+(defun flycheck-help-echo-helper (window object pos)
+  "Construct a tooltip message.
+
+Most of the actual work is done by calling
+`flycheck-help-echo-function' with the appropriate list of
+errors.  Arguments WINDOW, OBJECT and POS are as described in
+info node `(elisp)Special properties', as this function is
+intended to be used as the 'help-echo property of flycheck error
+overlays."
+  (-when-let (buf (cond ((bufferp object) object)
+                        ((overlayp object) (overlay-buffer object))))
+    (with-current-buffer buf
+      (-when-let* ((fn flycheck-help-echo-function)
+                   (errs (flycheck-overlay-errors-at pos)))
+        (funcall fn errs)))))
+
+(defun flycheck-make-help-echo (errs)
+  "Concatenate error messages and ids from ERRS.
+
+This function is the default for `flycheck-help-echo-function'."
```

Comment:
Please remove this line.  We should document the default at a single place, namely in `flycheck-help-echo-function`, to proactively avoid confusion if we ever change the default and accidentally forget to update this docstring.